include/SkirtBrim.h

```diff
@@ -186,6 +198,16 @@ class SkirtBrim
      */
     void generateSecondarySkirtBrim(Polygons& covered_area, std::vector<Polygons>& allowed_areas_per_extruder, std::vector<coord_t>& total_length);
 
+    /*!
+     * Generate the allowed areas for each extruder. Allowed areas represent where the brim/skirt is allowed to grow
+     * while adding new outset lines. This is basically the whole build plate, removing the areas where models are
+     * located, offsetted with some specific margins.
+     *
+     * \param[in] starting_outlines The previously generated starting outlines for each extruder
+     * \return The list of allowed areas for each extruder
+     */
+    std::vector<Polygons> generateAllowedAreas(const std::vector<Polygons>& starting_outlines) const;
```

Comment:
I have an inkling that this functionality is already present somewhere, did you double check this is not duplicated code?

Reply:
This is actully a specific part of code that was done in the `generate` function, but as I had to add some lines, I preferred to move it to a dedicate function.